app/src/main/java/com/sopt/androidstudy/presentation/save/screens/FriendGithubActivity.kt

```diff
@@ -29,28 +31,76 @@ class FriendGithubActivity : AppCompatActivity() {
     lateinit var user: String
     var isEnabled = true
     val friendGithubViewModel: FriendGithubViewModel by viewModels()
+
+    //For Thread
+    var i = 0
+    val countThread = thread(start = false) {
+        while (true) {
+            i++
+            val bundle = Bundle()
+            val message = myHandler.obtainMessage()
+            SystemClock.sleep(1000)
+            bundle.putInt("count", i)
+            message.data = bundle
+            message.what = COUNT_WHAT
+            myHandler.sendMessage(message)
+        }
+    }
+
+    val myHandler = MyHandler()
+    val backgroundHadlerThread = HandlerThread("HandlerThread")
+    lateinit var myThreadHandler: MyThreadHandler
+    //val background1Thread = Background1Thread(myHandler)
+
+
+    //lateinit var socket: Socket
+    //val myThread: ExecutorService = Executors.newFixedThreadPool(8)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         user = intent.getStringExtra("username").toString()
+        //ServerThread().start()
         binding = DataBindingUtil.setContentView(this, R.layout.activity_friend_github)
         user.let {
             friendGithubViewModel.setUserName(user)
             lifecycleScope.launch {
                 repeatOnLifecycle(Lifecycle.State.STARTED) {
                     friendGithubViewModel.getUserData.collect {
+                        Log.d("Collect!!", it.toString())
                         binding.user = it
                     }
                 }
             }
         }
+        backgroundHadlerThread.start()
+        myThreadHandler = MyThreadHandler()
+        binding.imageView.setOnClickListener {
+            if (countThread.isAlive) {
+                i = 0
+                Log.d("is Alive", countThread.toString())
+                countThread.interrupt()
+                if (!countThread.isAlive) {
+                    countThread.start()
+                }
+            } else {
+                countThread.start()
+            }
+        }
         bindingView()
+        friendGithubViewModel.selectUser.observe(this) {
+            val message = myThreadHandler.obtainMessage()
+            val bundle = Bundle()
+            bundle.putString("string", it.avatar_url)
+            message.data = bundle
+            message.what = IMAGE_WHAT
```

Comment:
```suggestion
            val bundle = Bundle()
            bundle.putString("string", it.avatar_url)
            val message = myThreadHandler.obtainMessage().apply {
                        data = bundle
                        what = IMAGE_WHAT
            }
```
이건 그냥 스킵하셔도 괜찮습니다! 이렇게도 바꿔볼 수 있을 것 같아요~!

Reply:
스킵하지 않아요~ 좋습니다